### PR TITLE
NDI Support

### PR DESCRIPTION
**Pull request by rustyclarksonfm**
Add NDI capture/render support to LiveSwitch Connect.
- only Windows & Mac supported
- audio is 16-bit
- video format support is currently limited (render = I420, capture = RGBA, BGRA)

### CODE REVIEW

**Review by ajones-frozenmountain — COMMENTED** [review]
The biggest update since the last review is the performance timer + linux libs. These changes look fine.

I'm still conscious this requires a 3rd party download, and would need to see how we want to handle that RE: licensing. 

Also, currently support video/audio types are small. is this something we want to flag more prominently in the Readme under NDI? This could save users time in downloading and applying their formats only to receive 'not supported'.

Codacy has picked up some issues in Receiver.cs

**Review by ajones-frozenmountain — APPROVED** [review]

**Review by amcgowan-frozenmountain — APPROVED** [review]

File: FM.LiveSwitch.Connect.sln [hunk 1]
```diff
@@ -5,16 +5,42 @@ VisualStudioVersion = 16.0.29519.87
...
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NDILibDotNet2", "src\NDILibDotNet2\NDILibDotNet2.csproj", "{B7874A82-4E4D-46D4-97DA-D28A46359CF0}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
+		Debug|x86 = Debug|x86
 		Release|Any CPU = Release|Any CPU
+		Release|x64 = Release|x64
+		Release|x86 = Release|x86
```
Comment (avenema) [inline FM.LiveSwitch.Connect.sln:17]: @rustyclarksonfm do we need to add these extra configurations? I am thinking that we only need `Any CPU` unless there are native 32-bit or 64-bit dependencies in here.

File: README.md [hunk 2]
```diff
@@ -13,13 +13,13 @@ Use `dotnet publish` to create a single, self-contained file for a specific plat
...
 ```shell
-dotnet publish -r win-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true -o win
+dotnet publish src/FM.LiveSwitch.Connect -r win-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true -o win
 ```
 
 ### macOS
 
 ```shell
-dotnet publish -r osx-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true -o osx
+dotnet publish src/FM.LiveSwitch.Connect -r osx-x64 -c Release /p:PublishSingleFile=true /p:PublishTrimmed=true -o osx
```
Comment (avenema) [inline README.md:22]: Need this for Linux as well, then.

File: src/FM.LiveSwitch.Connect/FM.LiveSwitch.Connect.csproj [hunk 3]
```diff
@@ -6,6 +6,11 @@
     <AssemblyName>lsconnect</AssemblyName>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <PlatformTarget>x64</PlatformTarget>
+  </PropertyGroup>
```
Comment (avenema) [inline src/FM.LiveSwitch.Connect/FM.LiveSwitch.Connect.csproj:12]: This isn't quite correct. A configuration of "Any CPU" shouldn't be targeting a specific platform. I think perhaps we want to drop "Any CPU" from the project/solution configurations and just go with "x64" as the required platform.

Comment (rustyclarksonfm) [reply src/FM.LiveSwitch.Connect/FM.LiveSwitch.Connect.csproj]: `Any CPU` worked after removing all of the x64/x86-specific configuration items. They were added when Spout was a dependency, but it was removed later and the project/solution files weren't properly updated.

**Pull request merged**